plugins/by-name/tardis/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, ... }:
+
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "tardis";
+  moduleName = "tardis-nvim";
+  packPathName = "tardis.nvim";
+  package = "tardis-nvim";
+  description = ''
+    Timetravel for neovim
+  '';
+
+  maintainers = [ lib.maintainers.fredeb ];
+
+  settingsOptions = {
```

Comment:
Generally, in recent code, we've not been adding new `settings` options.

That's not to say we have a hard rule against them, but usually we find they're more trouble than they're worth and make the plugin module harder to maintain.

Instead, we usually prefer having a `settingsExample` that gives an example of how to define `settings`. This could be copied from the `settings` value in your `example` test, for example.

Reply:
Okay, so just to make sure I understand, when a plugin has a `settings` field set in a users config, it that will be tranformed to a lua object that will be passed to the setup function for the plugin?

Reply:
Yes. `settings` gets serialized to lua, and passed to the plugin's `setup` function.